src/network/network_gui.cpp

```diff
@@ -1785,6 +1786,10 @@ struct NetworkClientListWindow : Window {
 				break;
 			}
 
+			case WID_CL_CLIENT_COMPANY_REMAINING:
+				SetDParam(0, _settings_client.network.max_companies - Company::GetNumItems());
```

Comment:
This setting does not get updated from the server. So, it will not be right when joining the server and whenever the server changes it won't be right. Unless "accidentally" the configuration value at the server happens to be the same as for your client, which is quite likely when testing it locally.

I would suggest you look at `NetworkMaxCompaniesReached` and add a similar function at those locations that returns the number of companies that are remaining. I would then also suggest changing `NetworkMaxCompaniesReached` so it uses that new function and compares that count with 0. That way the logic that chooses between the variables containing the maximum number of companies does not get duplicated and if the logic requires changes it is less likely that the other function is missed.